srtcore/core.cpp

```diff
@@ -11025,6 +11148,19 @@ SRT_REJECT_REASON CUDT::rejectReason(SRTSOCKET u)
     return s->m_pUDT->m_RejectReason;
 }
 
+int CUDT::rejectReason(SRTSOCKET u, int value)
+{
+    CUDTSocket* s = s_UDTUnited.locateSocket(u);
+    if (!s || !s->m_pUDT)
+        return APIError(MJ_NOTSUP, MN_SIDINVAL);
+
+    if (value < SRT_REJC_SERVER)
+        return APIError(MJ_NOTSUP, MN_INVAL);
+
+    s->m_pUDT->m_RejectReason = value;
```

Comment:
SRT_REJC_SERVER = 1000 (see srt.h)

srt_setrejectreason(..) should return error if the value >=2000 is passed.
The values below are SRT internal.

`if (value < SRT_REJC_SERVER)` does not protect from this.

Reply:
Or I am probably confused by SRT API and SRT protocol error ranges. 🤔

Reply:
There is a confusion, so let me explain:

The rejection code is passed through the URQ field in the handshake. For this number the ranges are valid codes, below 1000. Since 1000 on there start values of rejection codes, system ones this time, and those are these collected in SRT_REJECT_REASON. The code is then 1000 + the reject reason. Therefore for this field the server and user codes will be shifted by 2000 and 3000 respectively.

Translation from URQ into the rejection code is URQ - 1000, however:
* Codes from the system region (0 to 999) result in the system error code, unless the value is outside the system rejection values, in which case it renders to UNKNOWN
* Otherwise pass the code as is, just doing URQ - 1000.